pkg/minishift/registration/redhat.go

```diff
@@ -78,3 +80,26 @@ func (registrator *RedHatRegistrator) Unregister(param *RegistrationParameters)
 	}
 	return nil
 }
+
+//Put required proxy information to subscription-manager config file
+func (registrator *RedHatRegistrator) RegistrationManagerConfig(proxyInfo string) error {
+	if matched, _ := regexp.MatchString("http://", proxyInfo); matched {
+		proxyInfo = strings.TrimPrefix(proxyInfo, "http://")
+	} else if matched, _ := regexp.MatchString("https://", proxyInfo); matched {
+		fmt.Println("Removing https")
+		proxyInfo = strings.TrimPrefix(proxyInfo, "https://")
+	}
+
+	//Value of proxyInfo at this time would be similar to fedora:fedora123@xyz.com:8000
+	//Replacing the "@" with ":" and then splitting it with ":" will keep required values in a single slice
+	proxyInfo = strings.Replace(proxyInfo, "@", ":", 1)
+	proxyInfoSlice := strings.Split(proxyInfo, ":")
+
+	cmd := "sudo subscription-manager config"
+	cmd = cmd + fmt.Sprintf(" --server.proxy_hostname=%s --server.proxy_port=%s", proxyInfoSlice[2], proxyInfoSlice[3])
```

Comment:
I have handled it here https://github.com/minishift/minishift/pull/431/files#diff-fbcb3054efaea57dae48055b2d97af92R54 . The if statement will make sure it is only called when proxy is set.